### ISSUE TITLE
Obtain colors from os.proc.spawn stdout

### ISSUE THREAD

**Issue by peacefulotter** [com-lihaoyi/os-lib#214]
Hi,

I would like to get the colors from a subprocess stdout (e.g. `git status` will show the "modified" lines in green). From what I have seen, there is no way to do that in os-lib, maybe even in Java in general.

I have tried the following:
```scala
// ReadBytes or ReadLines have been tested
def processStdout = os.ProcessOutput.ReadBytes((line, len) =>
  val buf = line.slice(0, len).map(_.toChar)
  println(s"'${buf.mkString("Array(", ", ", ")")}'")
)
os.proc(cmd).spawn(stdout = processStdout) 
```
In this example, the stdout is printed as an array of Char but it is missing the color info such as `[92m some text here [0m`

Any idea how to do such a thing?  
Thanks!

**Comment by SethTisue** [comment]
Perhaps the subprocess is detecting that it isn't connected to the terminal, and is suppressing the color codes itself?

I don't know what `cmd` is but if it's e.g. `ls` you might need to e.g. `ls --color=always`

demo, not involving a JVM at all:

<img width="404" alt="Screenshot 2023-09-19 at 4 15 03 PM" src="https://github.com/com-lihaoyi/os-lib/assets/161079/381f549a-d1cc-4b54-a752-197b65b5e4ad">

**Comment by lefou** [comment]
If you emulate a terminal, you might want to set the `TERM` variable to something that's supposed to support colors, like `xterm-color`. Of course, you need to check which other implication that might have, e.g. what expectations do have commands you run to your emulated terminal then.

**Comment by peacefulotter** [comment]
Thank you, will try it out tonight

**Comment by peacefulotter** [comment]
Following up on this, I did the following:
```scala
os.proc(cmd :: params).call(
  cwd = wd,
  env = Map("TERM" -> "xterm-color"), // tried also just "xterm"
  stdout = processStdout
)
```
But I am still not getting the color codes...

And to make matters worse, the pty4j library can't be installed with SBT - at least it threw an error during installation.

I am on Windows 10 by the way, sorry for mentioning this so late.

**Comment by lefou** [comment]
Huh, what kind of update? I though, you are the one who wants to emulate a terminal? I guess you need to ispect at least one of the tools you want colored output of, and find out, on what criteria it decides to disable color. After knowning what's needed, we can think about how to provide it via os-lib.

**Comment by SethTisue** [comment]
I don't think this is an issue with os-lib. I believe you would experience the exact same issue using, for example, `java.lang.Process`, which is the underlying API that os-lib wraps.

I don't fault you for hoping we might have further advice for you, but regardless, I don't think this ticket should remain open, and I'm afraid you may need to seek advice elsewhere, since the os-lib bug tracker is focused on issues with os-lib itself.
